extensions/objects/templates/filepath-editor.directive.ts

```diff
@@ -762,7 +770,40 @@ angular.module('oppia').directive('filepathEditor', [
               AlertsService.addWarning('Could not get resampled file.');
               return;
             }
-            ctrl.postImageToServer(dimensions, resampledFile);
+            ctrl.saveImage(dimensions, resampledFile);
+          }
+        };
+
+        ctrl.saveImageToLocalStorage = function(
+            dimensions, resampledFile, imageType) {
+          var filename = ImageUploadHelperService.generateImageFilename(
+            dimensions.height, dimensions.width, imageType);
+          var reader = new FileReader();
+          reader.onload = function() {
+            var imageData = reader.result;
+            window.localStorage.setItem(filename, imageData);
```

Comment:
I would suggest creating a service for this which will handle upload/pop/flush/checking upload limit etc. (And we can access it through the entity directive for POST request.)

Reply:
Ok, will add a service.

Reply:
Done